include/nil/actor/zk/commitments/batched_commitment.hpp

```diff
@@ -104,7 +104,26 @@ namespace nil {
                         return math::lagrange_interpolation(U_interpolation_points);
                     }
 
-                    std::vector<std::vector<typename field_type::value_type>> get_unique_points_list() const{
+                    // We call them singles in recursive verifier
+                    std::vector<typename field_type::value_type> get_unique_points(){
+                        std::vector<typename field_type::value_type> result;
+//                        std::unordered_set<typename field_type::value_type> result_set;
+
+                        for( auto const &[k, point_batch]:_points ){
+                            for( auto const &point_set: point_batch ){
+                                for( auto const &point:point_set ){
+                                    if( std::find(result.begin(), result.end(), point) == result.end() ){
+                                        result.push_back(point);
+  //                                      result_set.insert(point);
+                                    }
+                                }
+                            }
+                        }
+
+                        return result;
+                    }
+
+                    std::vector<std::vector<typename field_type::value_type>> get_unique_point_sets_list() const{
```

Comment:
We can also use set here and don't need for loop inside to check was point in our list.